### PR TITLE
docs: update best practices

### LINKED ISSUE #16100: Create `config:best-practices` preset
### What would you like Renovate to be able to do?

We, as Renovate developers, want to have a way to tell our users: "here's what we recommend you do". A `config:best-practice` preset will help us do this.

Users can choose to extend from our `config:best-practice` preset. Nobody is "forced" into a workflow they may not like.

### If you have any ideas on how this should be implemented, please tell us here.

**Todo by Renovate programmer:**

- [x] Create new `config:best-practices` preset, which includes:
  - [x] Pin GitHub Actions to SHA
  - [x] Pin Docker digests
  - [x] Pin all `devDependencies`
  - [x] Any other important best-practice that the Renovate team recommends

**Todo by @HonkingGoose:**

- [x] Explain the new `config:best-practices` preset in the new best practices upgrade file in PR #22233
  - [x] Explain why we created the `config:best-practice(s)` preset
  - [x] Explain the reason behind each config option
  - [x] How to use our presets (put it in your `extends` array)
  - [x] Link to the preset config on the published docs

**Related issues:**

- #12024
- #11987

### Is this a feature you are interested in implementing yourself?

Partially, I'll let somebody else do the code work, and I'll do the documentation work.

### LINKED ISSUE #21783: Deep dive article: "Recommended upgrade practices from the Renovate developers"
### Describe the proposed change(s).

New deep dive article that boils down to "Here's what we Renovate developers recommend you do".

The recommendations should cover these situations:

- [x] new project
- [x] one year old project
- [x] five year old project
- [x] how to get your team to apply updates
- [x] how we use Renovate to update our own stuff

Also:

- [x] Add link to this article from the #21782 page
- [x] Mention the `config:best-practices` preset #16100
- [x] Link to the Swissquote user story

I'll start by writing a rough draft that the maintainers can then fill out with review suggestions.

### PR DESCRIPTION
**Pull request by HonkingGoose**
@coderabbitai: ignore

## Changes

- Our recommendations for updating dependencies
- Technical side and human side
- Different scenarios:
  - Fresh project
  - One year old project
  - Five year old project
- Explain the new `config:best-practices` preset

## Context

Closes: #21783
Closes: #16100

## Todos:

- [x] Get initial feedback from maintainers
- [x] Decide on filename and location
- [x] Use relative links for Renovate docs pages, after we decided on the filename and location
- [x] Put page in sidebar file
- [x] Link to this page from the "Reading list" page
- [x] Link to Swissquote user story
- [x] Mention `config:base` and `config:best-practices` presets somewhere suitable

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/upgrade-recommendations.md
Comment (morremeyer) [inline docs/usage/upgrade-recommendations.md:123]:
```suggestion
Focus on the process, not on the people.
```

File: docs/usage/upgrade-recommendations.md
Comment (morremeyer) [inline docs/usage/upgrade-recommendations.md:127]: I would focus on the things that help convince people to update - ideally dealing with a team that doesn't update for whatever reason is simply showing them that it is beneficial for everyone by being a role model. This enables these teams to look at what you're doing and then adapt these things for their process. 

Showing them how easy it is will make them want these easy processes, too. At least, that's my experience.

File: docs/usage/upgrade-recommendations.md
Comment (HonkingGoose) [reply docs/usage/upgrade-recommendations.md:24]: I think there's a spectrum of manual/automated updates, rough sketch:

- manually update with `npm update` when you remember it
- create checklist with timers/reminders to run `npm update` at set interval
- get bored of manual work, find a bot like Renovate
- get breaking update from the bot and get annoyed
- write test to cover the stuff that tends to break
- if tests pass: merge manually, if fail: find out why
- write more tests to cover more
- automerge updates when tests pass

So you start by doing everything yourself, and move further down the list to more automation. 

It's also okay to keep merging by hand, even if you have a full test suite, so you'll actually read the changelogs for the update. 😄

File: docs/usage/upgrade-recommendations.md
Comment (HonkingGoose) [reply docs/usage/upgrade-recommendations.md:127]: How easy updates are also depends on the quality and coverage of tests. To make updating as easy as it should be you need:

- Renovate bot
- Dependency Dashboard
- Right team mentality
- Company understands _why_ updating is generally safer than holding back update
- Tests for the critical path of the program

Updating _can_ be easy, but you have to work hard to get to the easy stuff. 😄

File: docs/usage/upgrade-recommendations.md
Comment (viceice) [inline docs/usage/upgrade-recommendations.md:19]:
```suggestion
If you think Renovate is too noisy, please read our [noise reduction docs](./noise-reduction.md).
```
isn't that file in same directory? so we should use relative link?

File: docs/usage/upgrade-recommendations.md
Comment (HonkingGoose) [inline docs/usage/upgrade-recommendations.md:28]:
```suggestion
- You'll be ready for CVE patches
- You'll look for ways to automate the update process
```

File: docs/usage/upgrade-recommendations.md
Comment (rarkins) [inline docs/usage/upgrade-recommendations.md:13]:
```suggestion
- Use the Dependency Dashboard issue (it's on by default)
```

**Review by rarkins — CHANGES_REQUESTED** [review]

File: docs/usage/upgrade-recommendations.md
Comment (viceice) [inline docs/usage/upgrade-recommendations.md:128]: 🤔

Comment (HonkingGoose) [reply docs/usage/upgrade-recommendations.md:158]: I still need suggestions from the Renovate maintainers on our recommendations to deal with the team dynamics. 😉

File: docs/usage/upgrade-best-practices.md
Comment (HonkingGoose) [inline docs/usage/upgrade-best-practices.md:21]:
```suggestion
## The `config:best-practices` preset

Short introduction goes here.

### Why we created the `config:best-practices` preset

Explain the reason behind each config option in the preset.
How to use the preset (extend it).
Link to the config preset in the published docs.

```

File: docs/usage/upgrade-best-practices.md
Comment (viceice) [inline docs/usage/upgrade-best-practices.md:215]:
```suggestion
- Enable [GitHub Merge Queue](./key-concepts/automerge.md#github-merge-queue) to speed up merges
```

this is another test error

File: docs/usage/upgrade-best-practices.md
Comment (viceice) [inline docs/usage/upgrade-best-practices.md:121]:
```suggestion
You may also start using [GitHub's pull request merge queues](./key-concepts/automerge.md#github-merge-queue) to speed up the merge process.
```

This is one test error.

File: docs/usage/upgrade-best-practices.md
Comment (viceice) [reply docs/usage/upgrade-best-practices.md:1]: see comments below 😉

File: docs/usage/upgrade-best-practices.md
Comment (secustor) [inline docs/usage/upgrade-best-practices.md:169]: We could reference our vulnerabilitiy options here

Comment (HonkingGoose) [reply docs/usage/upgrade-best-practices.md:169]:
```suggestion
First update any dependencies that have critical updates for CVEs or other security related improvements.

Follow the steps listed in the [`vulnerabilityAlerts`](https://docs.renovatebot.com/configuration-options/#vulnerabilityalerts) docs to make sure Renovate is reading the GitHub Vulnerability Alerts.
```

Is this what you mean? I'm not quite sure if you _must_ have `vulnerabilityAlerts` set up to get CVE fixes, or if this is just about customizing the CVE fixes you would get anyways...

Comment (secustor) [reply docs/usage/upgrade-best-practices.md:169]: It has to be set up and only works for Github. 

What I reffered to was `osvVulnerabilities` too

**Review by HonkingGoose — COMMENTED** [review]
I'm ready for a round of review, at last. 😄

**Review by secustor — DISMISSED** [review]

File: docs/usage/upgrade-best-practices.md
Comment (jamietanna) [inline docs/usage/upgrade-best-practices.md:93]: I realise we've already decided on what `config:best-practices` contains, but is it worth having major and minor split out? Means it's easier to apply `major` when you're ready, while retaining ability to do minor/patch

File: docs/usage/upgrade-best-practices.md
Comment (secustor) [reply docs/usage/upgrade-best-practices.md:62]: Yes, as it get's harder for users to change configs when the their config deviates with the current config format.

File: docs/usage/upgrade-best-practices.md
Comment (HonkingGoose) [reply docs/usage/upgrade-best-practices.md:95]: Our [FAQ, default behavior for major/minor releases](https://docs.renovatebot.com/faq/#renovates-default-behavior-for-majorminor-releases) says:

> By default, Renovate separates major and minor releases.

So, just going of that, Renovate should already do what you want? It's not clear to me if we mean "default behavior if no config at all", or "default behavior for `config:recommended`". I'm probably missing something important, because you're asking for a behavior change... 😄

File: docs/usage/upgrade-best-practices.md
Comment (rarkins) [reply docs/usage/upgrade-best-practices.md:276]: We can keep them in

**Review by rarkins — DISMISSED** [review]

File: docs/usage/upgrade-best-practices.md
Comment (viceice) [inline docs/usage/upgrade-best-practices.md:189]: add a note to separate multiple major versions config option?

**Review by viceice — COMMENTED** [review]
otherwise LGTM

**Review by viceice — DISMISSED** [review]

**Review by secustor — DISMISSED** [review]

File: docs/usage/upgrade-best-practices.md
Comment (lmilbaum) [inline docs/usage/upgrade-best-practices.md:18]: strategy or stategy?

File: docs/usage/upgrade-best-practices.md
Comment (TWiStErRob) [inline docs/usage/upgrade-best-practices.md:76]: @HonkingGoose I just read a really nice real-life example for why we want this:
https://www.paloaltonetworks.com/blog/prisma-cloud/github-actions-worm-dependencies/

Comment (HonkingGoose) [reply docs/usage/upgrade-best-practices.md:76]: Thank you for the article! This is the basic attack:

> We can use the secrets exfiltrated in the flow to infect the repository with malicious code. Overwriting a commit while keeping its hash the same isn’t possible, so we can’t abuse a commit hash reference.

And they recommend:

> 4. Pin actions using a commit hash to reduce the risk of using a maliciously modified action.

Comment (TWiStErRob) [reply docs/usage/upgrade-best-practices.md:76]: Yep, basic, which means it's really easy, and easy to protect against too. :)

Comment (HonkingGoose) [reply docs/usage/upgrade-best-practices.md:76]: We could link to the article, or keep things the same.

In favor of the article would be that it explains why pinning to the commit hash is good, and it helps strengthen our "pin all actions"-stance.

Against the article is that it may be too much information.

I'll let the maintainer decide. 😉

Comment (secustor) [reply docs/usage/upgrade-best-practices.md:76]: I'm fine with linking to the blog if the user wants some more detailed information. 

Something like: 
```
For an in depth explanation why you should pin your Github Actions see: 
```
or something similar.

**Review by viceice — DISMISSED** [review]

**Review by rarkins — APPROVED** [review]

**Review by secustor — APPROVED** [review]

**Pull request merged**
